### PR TITLE
[BugFix] fix incorrect name when fetch data in sparse optim

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

fix for m5gnn
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/distributed/optim/pytorch/sparse_optim.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def step(self):
                 # will send grad to each corresponding trainer
                 if self._world_size > 1:
                     # get idx split from kvstore
-                    idx_split = kvstore.get_partid(name, idics)
```
Comment (zheng-da) [inline python/dgl/distributed/optim/pytorch/sparse_optim.py:77]: you should remove `name = emb._tensor.name` at line 49

Comment (zheng-da) [reply python/dgl/distributed/optim/pytorch/sparse_optim.py:77]: i'm bit confused. `emb._tensor.name` and `emb.data_name` are different?

Comment (Rhett-Ying) [reply python/dgl/distributed/optim/pytorch/sparse_optim.py:77]: yes, they're different. as `group_id` is attached to the name user specifies. so `_tensor.name` returns the original name which does not append with `group_id`. `_tensor._name` is the globally unique name which is attached with `group_id`. It's confusing indeed.

Comment (zheng-da) [reply python/dgl/distributed/optim/pytorch/sparse_optim.py:77]: can you remove line 49?
can you add a test for this modification?

Comment (Rhett-Ying) [reply python/dgl/distributed/optim/pytorch/sparse_optim.py:77]: add test may be not possible, as the code logic can only be touched while `world_size` > 1 and distributed is enabled. Is it possible to add such a test in `test_dist_optim.py`?

Comment (Rhett-Ying) [reply python/dgl/distributed/optim/pytorch/sparse_optim.py:77]: and line 49 cannot be removed, it's used in line 114/118/120/121

**Review by zheng-da — APPROVED** [review]

**Pull request merged**